src/DotFlyer.Service/Senders/EmailSender.cs

```diff
@@ -22,37 +22,57 @@ public class EmailSender(
     /// <exception cref="HttpRequestException">Thrown when the email message could not be sent.</exception>"
     public async Task SendAsync(EmailMessage emailMessage, CancellationToken cancellationToken = default)
     {
-        if (emailMessage.To.Count == 0)
+        if (!emailMessage.To.Any())
         {
             throw new ArgumentException("Email message must have at least one recipient in the 'To' field.");
         }
 
         SendGridMessage sendGridMessage = new()
         {
-            From = new(emailMessage.FromEmail, emailMessage.FromName),
+            From = new(emailMessage.From.Email, emailMessage.From.Name),
             Subject = emailMessage.Subject,
             HtmlContent = emailMessage.Body
         };
 
-        emailMessage.To.ForEach(emailRecipient => sendGridMessage.AddTo(new EmailAddress(emailRecipient.Email, emailRecipient.Name)));
-        emailMessage.Cc.ForEach(emailRecipient => sendGridMessage.AddCc(new EmailAddress(emailRecipient.Email, emailRecipient.Name)));
-        emailMessage.Bcc.ForEach(emailRecipient => sendGridMessage.AddBcc(new EmailAddress(emailRecipient.Email, emailRecipient.Name)));
-
-        foreach (var attachment in emailMessage.Attachments)
+        foreach (var contact in emailMessage.To)
         {
-            BlobClient blobClient = new(new(attachment), credential);
+            sendGridMessage.AddTo(new EmailAddress(contact.Email, contact.Name));
+        }
 
-            if (await blobClient.ExistsAsync(cancellationToken))
+        if (emailMessage.Cc != null)
+        {
+            foreach (var contact in emailMessage.Cc)
             {
-                using MemoryStream blobStream = new();
-
-                await blobClient.DownloadToAsync(blobStream, cancellationToken);
+                sendGridMessage.AddCc(new EmailAddress(contact.Email, contact.Name));
```

Comment:
as discussed on the phone, we can consider implicit casting operator between Contact and EmailAddress 

just a suggestion

i.e. https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/user-defined-conversion-operators

Reply:
never mind, you do not have  - and ideally should not have - dependency on sendgrid from common

moot point, resolving this comment